### ISSUE TITLE
Add MakeFile to ease development process 

### ISSUE THREAD

**Issue by PradyumnaKrishna** [IndianOpenSourceFoundation/dynamic-cli#162]
### 🚀 Feature

Create a `MakeFile` to simplify project setup and ease the development process.
It requires #161 to complete, so pickup that or wait for it to complete before start working on this.

### Resources
1. https://makefiletutorial.com

Feel free to contact me for help.

**Comment by shreelakshmijoshi** [comment]
Hi @PradyumnaKrishna! Would you assign me this issue under GSSOC?

**Comment by PradyumnaKrishna** [comment]
Done @shreelakshmijoshi

**Comment by dhruti29** [comment]
Hello, @PradyumnaKrishna I want work on this issue please assign me this issue under GSSOC...!!

**Comment by GouravSardana** [comment]
@dhruti29 , Please go ahead

**Comment by shreelakshmijoshi** [comment]
Hi @GouravSardana !
I want to understand a bit more about Make files. What exact dependencies/ file names should we include in the Make file to compile them?
Also, in the [make file resource](https://makefiletutorial.com/) provided, it is mentioned 
> Interpreted languages like Python, Ruby, and Javascript don't require an analogue to Makefiles.

And dynamic-cli has all Python files.

**Comment by GouravSardana** [comment]
@PradyumnaKrishna is leading the effort from this side. Can you please explain what we need from our side ?

**Comment by GouravSardana** [comment]
I agree with the point which you mentioned but let’s hear from @PradyumnaKrishna

**Comment by PradyumnaKrishna** [comment]
Makefile does lot more than just compiling, currently this repository has lot of commands that can be aggregated at one place.

Let me give you an example

```Makefile
all: setup lint test build

setup:
	@echo SETUP
	pip install -r requirements.txt

lint:
	@echo LINT
	# Linting it using linters like black, pylint etc

test:
	@echo TEST
	pytest

build:
	@echo BUILD
	...

publish:
	@echo PUBLISH
	...
...
```

These all tasks and other similar tasks will get collected to one place. Currently we have to follow up the procedure to setup the development environment. After implementing Makefile, environment setup and package build will get done in one go, just have to type `make`.

Note: #161 is work in progress which add poetry as dependency manager, the procedure to setup the development environment requires some changes after that.

**Comment by shreelakshmijoshi** [comment]
@PradyumnaKrishna, yes you're right.

**Comment by harshsingh32** [comment]
I would like to work on this issue, please assign this issue to me under GSSoC'23.
